tests/Doctrine/Tests/TestUtil.php

```diff
@@ -61,16 +61,28 @@ private static function hasRequiredConnectionParams()
             $GLOBALS['db_type'],
             $GLOBALS['db_username'],
             $GLOBALS['db_password'],
-            $GLOBALS['db_host'],
-            $GLOBALS['db_name'],
-            $GLOBALS['db_port']
+            $GLOBALS['db_name']
+        )
+        && (
+            isset(
+                $GLOBALS['db_host'],
+                $GLOBALS['db_port']
+            ) || isset(
+                $GLOBALS['db_server']
```

Comment:
@Deltachaos I get that but this class is meant to be able to deal with all vendor connection. To me it sounds as if `server` is actually `host` (Doctrine wise; abstracted) because how can you simply connect to a database server only providing an arbitrary "server name" without the client knowing the host address? Or is the concept like ODBC where you define a data source name (`server`) which then resolves params like `host`, `port` etc.? Because then I would map Sybase's `server` param to Doctrine's `host` param in the driver. This is the most convenient way IMO (see Doctrine 1 with `dblib` driver for example). We cannot manage all the custom vendor implementation connection parameters and need to find common denominator for the most crucial ones. Just have a look at the Oracle drivers and you'll get the idea.

Hope this helps (tending to close this as this seems to be a misunderstanding of how DBAL works and need driver fixing).